src/adapter/graphql/src/queries/datasets/dataset.rs

```diff
@@ -17,16 +17,16 @@ use crate::queries::*;
 
 #[derive(Debug, Clone)]
 pub struct Dataset {
-    account: Account,
+    owner_ref: AccountRef,
```

Comment:
What's the motivation for this change of using `AccountRef` instead of `Account` everywhere?

It seems to break the whole "Graphness" of GQL:
- Previously you could potentially traverse nodes infinitely through connections like `dataset -> owner -> organizations[0] -> teams[0] -> datasets[0] -> owner ....`.
- But now `AccountRef` is a dead-end node with just an ID and Name, which is very REST-like.

In some comment previously I was suggesting to introduce `AccountRef`, but it was mostly for the ODF side of things - as a direct counterpart for `DatasetRef` that allows to refer to an account EITHER by its name or by ID, and then a `AccountHandle` which is a resolved reference that has BOTH name and ID. (this is of course out of scope of this PR).

Reply:
If I only have  a reference (we usually only have Name to be exact), anything else like avatar URL, not even talking of tracing organizations, needs to interact with `AuthenticiationService` to resolve full account info. This is expensive to precompute (remember, we are going to GitHub API right now - in future, a database round trip), and most of the time these attributes would not be needed much. Computing full account info on each GraphQL field resolution is also expensive. This would need some sort of lazy load wrapper, if it's possible at all in Rust with the immutability rules...

Reply:
Making an experiment with `tokio::sync::OnceCell` for lazy evaluation